src/tactics/FStar.Tactics.Hooks.fs

```diff
@@ -220,8 +220,8 @@ let rec traverse (f: pol -> Env.env -> term -> tres) (pol:pol) (e:Env.env) (t:te
             // TODO: traverse the types?
             comb1 (fun t -> Tm_ascribed (t, asc, ef)) (traverse f pol e t)
 
-        | Tm_match (sc, brs) ->
-            comb2 (fun sc brs -> Tm_match (sc, brs))
+        | Tm_match (sc, asc_opt, brs) ->  //AR: not traversing the return annotation
```

Comment:
Again, similar to the handling to Tm_ascribed, though perhaps both should descend into the annotation